File: introducing-go/ch06/swap.go
package main

import "fmt"

func swap(x *int, y *int) {
	*x, *y = *y, *x
}

func main() {
	x, y := 1, 2
	fmt.Printf("x = %d, and y = %d\n", x, y)
	swap(&x, &y)
	fmt.Printf("x = %d, and y = %d\n", x, y)
}

File: effective-c/ch01/swap.c
#include <stdio.h>
#include <stdlib.h>

void swap(int*, int*);

int
main(void)
{
	int a = 21;
	int b = 17;

	printf("before swap: a = %d, b = %d\n", a, b);
	swap(&a, &b);
	printf("after  swap: a = %d, b = %d\n", a, b);

	return EXIT_SUCCESS;
}

void
swap(int *a, int *b)
{
	int t = *a;
	*a = *b;
	*b = t;
	return;
}
